src/request.rs

```diff
@@ -10,10 +10,11 @@ pub fn with_session<T: SessionStore>(
     session_store: T,
     cookie_options: Option<CookieOptions>,
 ) -> impl Filter<Extract = (SessionWithStore<T>,), Error = Rejection> + Clone {
-    let cookie_options = match cookie_options {
+    let mut cookie_options = match cookie_options {
         Some(co) => co,
         None => CookieOptions::default(),
     };
+    cookie_options.cookie_name = "sid".to_owned();
```

Comment:
Good catch, thank you for the PR! I was manually specifying a session ID in my usage and didn't realize the default usage was breaking.

I think we should maintain the ability to rename the cookie however, and use the specified name in later logic. To be able to pass the name in, let's go ahead and make `cookie_name` an `&'static str` to match the filter param, i.e.
```rust
pub struct CookieOptions {
    pub cookie_name: &'static str,
    pub cookie_value: Option<String>,
    ...
}
```
This is actually quite nice as it removes our need to call `.to_string()` everywhere.

And then from there, let's only apply the default cookie_name = "sid" when building a default cookie because the parameter passed in was None. So something like this:
```rust
    let cookie_options = match cookie_options {
        Some(co) => co,
        None => {
            let mut co = CookieOptions::default();
            co.cookie_name = "sid";
            co
        }
    };
```